src/sql/parts/notebook/notebookActions.ts

```diff
@@ -48,6 +48,36 @@ export class AddCellAction extends Action {
 	}
 }
 
+export class SaveNotebookAction extends Action {
+	private static readonly notebookSavedMsg = localize('notebookSavedMsg', 'Notebook saved successfully.');
+	private static readonly notebookFailedSaveMsg = localize('notebookFailedSaveMsg', 'Failed to save Notebook.');
+	constructor(
+		id: string, label: string, cssClass: string,
+		@INotificationService private _notificationService: INotificationService
+	) {
+		super(id, label, cssClass);
+	}
+
+	public run(context: NotebookComponent): TPromise<boolean> {
+		return new TPromise<boolean>((resolve, reject) => {
+			try {
+				const actions: INotificationActions = { primary: [] };
+				if(context.save())
```

Comment:
You need to change this to an async method, then do
```
let saved = await context.save();
if (saved) {
...
```

Otherwise this will never return false as it's "truthy" (it returns a promise object).

Reply:
Done.